purchase_request/wizard/purchase_request_line_make_purchase_order.py

```diff
@@ -199,7 +199,7 @@ def _get_order_line_search_domain(self, order, item):
             ("name", "=", name),
             ("product_id", "=", item.product_id.id),
             ("product_uom", "=", vals["product_uom"]),
-            ("analytic_distribution", "=?", item.line_id.analytic_distribution),
+            ("analytic_distribution", "=", item.line_id.analytic_distribution),
```

Comment:
It is not a typo. `=?` is a valid operator: https://github.com/odoo/odoo/blob/16.0/odoo/osv/expression.py#L1120-L1122

But the analytic distribution is a tricky field, and I assume the operator may well be buggy in this case. To emulate the behaviour of the operator, you need to include this clause in the domain optionally (i.e. if `item.line_id.analytic_distribution` has any content).